bind.go

```diff
@@ -181,42 +184,27 @@ func (b *bindIncr[A, B]) unlinkBindChange(ctx context.Context) {
 
 func (b *bindIncr[A, B]) unlinkOld(ctx context.Context) {
 	if b.bound != nil {
-		TracePrintf(ctx, "%v unbinding node %v", b, b.bound)
+		TracePrintf(ctx, "%v unbinding old rhs %v", b, b.bound)
 		b.unlinkBindChange(ctx)
 		b.removeNodesFromScope(ctx, b.scope)
 		b.n.graph.unobserveNodes(ctx, b.bound, b.n.Observers()...)
 		for _, c := range b.n.Children() {
 			Unlink(c, b.bound)
 		}
 		b.bindChange = nil
-		b.scope = nil
 		b.bound = nil
 	}
 }
 
-func (b *bindIncr[A, B]) addNodesToScope(ctx context.Context, scope *bindScope, gn INode) {
-	gnn := gn.Node()
-	scope.rhsNodes.Push(gn)
-	parents := gnn.Parents()
-	for _, p := range parents {
-		b.addNodesToScope(ctx, scope, p)
-	}
-	if typed, ok := gn.(IBind); ok {
-		typed.Link(ctx)
-	}
-	gnn.createdIn[b.n.id] = scope
-}
-
 func (b *bindIncr[A, B]) removeNodesFromScope(ctx context.Context, scope *bindScope) {
 	rhsNodes := scope.rhsNodes.Values()
 	for _, n := range rhsNodes {
-		delete(n.Node().createdIn, b.n.id)
-		if len(n.Node().createdIn) == 0 {
-			if typed, ok := n.(IBind); ok {
-				typed.Unlink(ctx)
-			}
+		n.Node().createdIn = nil
+		if typed, ok := n.(IBind); ok {
+			typed.Unlink(ctx)
```

Comment:
I think we need to make sure rhsNodes contains only nodes created in the given scope. Not sure where in the code we guarantee this condition. It looks to me we just add any node returned by f. This is so that when given shared nodes created in some other scope, we don't unnecessarily invoke`unlink`. 

I run some complicated formulas against main and there were nil exceptions and my current theory from nature of the trace is that this weird behavior seem to be a result of unrelated bind nodes detaching themselves from n.bound in some later step after they stabilize. So that when other nodes that depend on these "mistakenly unbound" nodes stabilize, they end up working with nils. I'm currently trying to come up with a simple repro test case. But that's my intuition so far.

Reply:
That's exactly the thrust of this PR; basically we don't want to walk the parent lists up and add nodes seen that way, we want to add nodes only from the constructor (at construction) and only for the "innermost" bind, or the bind that called the closure most recently.

I have another test in the diff to assert that works correctly, though I can throw more test cases at it. 

For this specific test though, if you just walk the calls by hand you'll still spot the cycle because of how `f(N)` gets used in multiple places